core/log/file_console_logger.odin

```diff
@@ -31,44 +34,158 @@ Default_File_Logger_Opts :: Options{
 	.Procedure,
 } | Full_Timestamp_Opts
 
+@private
+File_Type :: enum {
+	console,
+	os,
+	os2,
+}
 
+/*
+	Implements a logger that writes to a file stream, such as stdout, or a file on the system.
+	
+	This logger is compatible both with `os` and `os2`.
+	- To make a logger that writes to an `os.Handle` refer to `make_file_logger_os`
+	- To make a logger that writes to an `^os2.File` refer to `make_file_logger_os2`
+	- To make a logger that writes to `stdout` and `stderr` refer to `make_console_logger`
+*/
 File_Console_Logger_Data :: struct {
-	file_handle:  os.Handle,
+	type: File_Type,
+	file_handle_os: os.Handle,
+	file_handle_os2: ^os2.File,
 	ident: string,
+	close_file_on_delete: bool,
+	allocator: runtime.Allocator,
 }
 
-create_file_logger :: proc(h: os.Handle, lowest := Level.Debug, opt := Default_File_Logger_Opts, ident := "") -> Logger {
-	data := new(File_Console_Logger_Data)
-	data.file_handle = h
+/*
+Makes a new logger that will write to the provided `os.Handle`
```

Comment:
A kind of a nitpick, but the first line of the documentation comment shouldn't describe what the procedure does, as in "Does X", but tells what it is doing in a more imperative way, e.g. "Do X". So for this procedure that would be

```
Make a new logger that will write to the provided Handle.
```

Also I'm not sure, if the usage of the word `make` is correct here per se. Make is used by the builtins that allow you to allocate data structures such as slices, dynamic arrays and maps. For singular objects, the words `new` and `delete` are used.